src/Draco.Compiler/Internal/Syntax/Parser.cs

```diff
@@ -198,7 +198,8 @@ public CompilationUnitSyntax ParseCompilationUnit()
     /// <returns>The parsed <see cref="DeclarationSyntax"/>.</returns>
     internal DeclarationSyntax ParseDeclaration()
     {
-        switch (this.Peek())
+        var peekAmount = this.Peek() == TokenKind.KeywordInternal || this.Peek() == TokenKind.KeywordPublic ? 1 : 0;
```

Comment:
You could add an `IsVisibilityModifier` to make the condition nicer, we have a bunch of these at the top